pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +501,66 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for ((domain, sender, message), _) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push((domain.clone(), sender.clone(), message.clone()));
+
+				let router = match DomainRouters::<T>::get(domain.clone()) {
+					Some(router) => router,
+					None => {
+						Self::deposit_event(Event::RouterNotFound {
+							domain: domain.clone(),
+						});
```

Comment:
If this error occurs, the message should be stored in an error storage with the problem that happened, or we we stop processing.

Reply:
If we check the router when submitting, this would mean that intermediately, the router has been removed, but still state changes on Centrifuge Chain happened. So an error storage is the best IMO.

Reply:
Saving the error along with the message info in the `FailedOutboundMessages` storage.